### DISCUSSION TITLE
Difference between openssl version 3.0.* vs 3.1.* vs 3.2.* vs 3.3.*

### DISCUSSION THREAD

**Discussion by aniljadaun** [openssl/openssl#24695] [General Discussion]
We are currently using openssl 3.0.10 for base and 3.0.8 for fips provider. There are some vulnerabilities (https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=3.0.10) reported for 3.0.8 and thus we  plan to update openssl 3.0 to latest version . Now i am confused what does this different version have in them
 3.0.* vs 3.1.* vs 3.2.* vs 3.3.*  .  Does 3.3.x version has significant implementation changes for earlier available interfaces , algorithm or it introduces only new implementation , features ?

**Comment by paulidale** [comment]
Look [here](https://github.com/openssl/openssl/blob/master/CHANGES.md) for the differences.

You shouldn't need to upgrade the FIPS provider from 3.0.8, the only vulnerability inside the provider only appears on ARM64 platforms using XTS mode.  Upgrading OpenSSL to 3.0.14 or later is sensible.

**Reply by aniljadaun** [reply]
Thanks for the guidance :)
